packages/cli/src/commands/hydrogen/link.test.ts

```diff
@@ -67,33 +75,132 @@ describe('link', () => {
     expect(getStorefronts).toHaveBeenCalledWith(SHOP);
   });
 
-  it('renders a list of choices and forwards the selection to setStorefront', async () => {
-    vi.mocked(renderSelectPrompt).mockResolvedValue(
-      'gid://shopify/HydrogenStorefront/1',
-    );
+  describe('when you want to link an existing Hydrogen storefront', () => {
+    beforeEach(async () => {
+      vi.mocked(renderSelectPrompt).mockResolvedValue(
+        'gid://shopify/HydrogenStorefront/1',
+      );
+    });
+
+    it('renders a list of choices and forwards the selection to setStorefront', async () => {
+      await linkStorefront({path: 'my-path'});
+
+      expect(setStorefront).toHaveBeenCalledWith(
+        'my-path',
+        expect.objectContaining({
+          id: 'gid://shopify/HydrogenStorefront/1',
+          title: 'Hydrogen',
+        }),
+      );
+    });
 
-    await linkStorefront({path: 'my-path'});
+    it('renders a success message', async () => {
+      await linkStorefront({path: 'my-path'});
 
-    expect(setStorefront).toHaveBeenCalledWith(
-      'my-path',
-      expect.objectContaining({
-        id: 'gid://shopify/HydrogenStorefront/1',
-        title: 'Hydrogen',
-      }),
-    );
+      expect(outputMock.info()).toMatch(/Hydrogen is now linked/g);
+      expect(outputMock.info()).toMatch(
+        /Run `h2 dev` to start your local development server and start building/g,
+      );
+    });
   });
 
-  it('renders a success message', async () => {
-    vi.mocked(renderSelectPrompt).mockResolvedValue(
-      'gid://shopify/HydrogenStorefront/1',
-    );
+  describe('when you want to link a new Hydrogen storefront', () => {
+    beforeEach(async () => {
+      vi.mocked(renderSelectPrompt).mockResolvedValue('NEW_STOREFRONT');
+    });
+
+    it('chooses to create a new storefront given the directory path', async () => {
+      await linkStorefront({path: 'my-path'});
+
+      expect(setStorefront).not.toHaveBeenCalled();
+      expect(renderTextPrompt).toHaveBeenCalledWith({
+        message: 'What do you want to name the Hydrogen storefront on Shopify?',
+        defaultValue: 'my-path',
+      });
+    });
+
+    it('chooses to create a new storefront without directory path', async () => {
+      await linkStorefront({});
+
+      expect(setStorefront).not.toHaveBeenCalled();
+      expect(renderTextPrompt).toHaveBeenCalledWith({
+        message: 'What do you want to name the Hydrogen storefront on Shopify?',
+        defaultValue: 'Hydrogen Storefront',
+      });
```

Comment:
Should i save the messages in a `const` or do you just mean accept any value for certain attributes.

Reply:
I mean it might be enough to just assert `...toHaveBeenCalledWith(expect.objectContaining({message: expect.stringMatching(/name/i)}))` or similar, instead of matching the whole string. That gives some flexibility to modify the prompt without breaking the test, since the main question there should always be about "name".